### PR TITLE
`rustdoc` or `cargo doc` should pass a special `--cfg doc` flag 

### PR DESCRIPTION
**Pull request by kennytm**
[Rendered](https://github.com/kennytm/rfcs/blob/rustdoc-flag/text/0000-cfg-rustdoc.md)

cc rust-lang/rust#19240

### CODE REVIEW

File: text/0000-cfg-rustdoc.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+Users can add documentation-only declarations with the `#[cfg(rustdoc)]` attribute.
+
+# Drawbacks
+
+Possibly abused to produce documentation not matching the actual API.
+
+# Alternatives
+
+* The identifier `rustdoc` can be changed to something else.
```
Comment (P1start) [inline text/0000-cfg-rustdoc.md]: I think this should just be `doc`. `rustdoc` sounds specific to the `rustdoc` tool itself, and so if someone made another documentation tool for Rust called something else, the `rustdoc` cfg attr wouldn’t really make much sense any more. A generic `doc` cfg sounds much more general.

Comment (blaenk) [inline text/0000-cfg-rustdoc.md]: Yeah, `doc` sounds better to me.

Comment (pcwalton) [inline text/0000-cfg-rustdoc.md]: +1 on `doc`.

**Comment by nrc** [comment]
This is probably a good idea for us to do. However, now is not the time - we are very close to the 1.0 release and this is neither urgent nor has backwards compatibility hazards. Therefore we decided it should be closed as postponed. I created #915 to track this issue for now.

**Comment by lfairy** [comment]
Should we re-open this RFC?

The proposal seems pretty straightforward, and we're already in the 1.6 cycle.

**Comment by nodakai** [comment]
Ping?

**Comment by mitaa** [comment]
I think this is a good idea, but this also means that there must not be any compile errors from any of the `#[cfg(doc)]`'d items (I imagine this might be a problem, requiring workarounds like dummy items?).

cc https://github.com/rust-lang/rust/issues/1998

**Pull request closed without merging**